### PR TITLE
Givinar/fix-pdf

### PR DESCRIPTION
**Pull request by givinar**
- Add uniform
- Fix pdf for NIS
- Map cartesian to spherical coord. system 
- ! return true determinant (need to check)

### CODE REVIEW

File: utils/utils.py [hunk 1]
```diff
@@ -213,21 +219,31 @@ def map_to_hemisphere(s: np.ndarray, n: np.ndarray, e=1.):
...
     norm_pts = points[:, 3:]
-    lights = np.zeros(norm_pts.shape, dtype=np.float32)
+    lights = np.zeros([norm_pts.shape[0], 2], dtype=np.float32)
     pdfs = np.zeros(norm_pts.shape[0], dtype=np.float32)
 
     for i in range(norm_pts.shape[0]):
         s = np.random.uniform(0., 1., 2)
-        norm = norm_pts[i]
-        light = map_to_hemisphere(s, norm)
+        norm_p = norm_pts[i]
```
Comment (LapshinA) [inline utils/utils.py]: Do we use norm_p here?

Comment (givinar) [reply utils/utils.py]: Nope. Deleted it.

File: integration_experiment.py [hunk 2]
```diff
@@ -165,17 +166,15 @@ def make_infer(self):
...
-            l = np.linalg.norm(samples, axis=-1)
-            v = np.abs(l - 1)
-            res = np.all(v <= np.finfo(np.float32).eps)
-            if not res:
-                logging.warning("Vector not equ 1")
-
         else:
             [samples, pdfs] = self.nis.get_samples(points)
+            samples[:, 0] = torch.acos(samples[:, 0])
+            samples[:, 1] = samples[:, 1] * 2 * math.pi
```
Comment (LapshinA) [inline integration_experiment.py]: np.pi no import math

File: integrator.py [hunk 3]
```diff
@@ -214,15 +220,21 @@ def train_with_context(self, context: np.ndarray, batch_size=100, lr=None, point
...
+            #mean = torch.mean(y / absdet)
+            mean = torch.mean(-log_absdet - log_y)
+            var = torch.var(y * absdet)
+            #var = torch.var(y / absdet)
+            #y = (y / mean).detach()
 
             # Backprop #
-            loss = self.loss_func(y, absdet)
+            #loss = self.loss_func(y, absdet)
+            loss = mean
```
Comment (LapshinA) [inline integrator.py:235]: Do we use mean instead of loss_func?

File: integrator.py [hunk 4]
```diff
@@ -214,15 +220,21 @@ def train_with_context(self, context: np.ndarray, batch_size=100, lr=None, point
 
             # --------------- START TODO compute loss ---------------
             y = batch_y.to(self.device)
-            mean = torch.mean(y / absdet)
-            var = torch.var(y / absdet)
-            y = (y / mean).detach()
+            y = y + np.finfo(np.float32).eps
+
+            log_y = torch.log(y)
```
Comment (LapshinA) [inline integrator.py:225]: This increasing computation graph.

File: utils/utils.py [hunk 5]
```diff
@@ -213,21 +219,29 @@ def map_to_hemisphere(s: np.ndarray, n: np.ndarray, e=1.):
...
     norm_pts = points[:, 3:]
-    lights = np.zeros(norm_pts.shape, dtype=np.float32)
+    lights = np.zeros([norm_pts.shape[0], 2], dtype=np.float32)
     pdfs = np.zeros(norm_pts.shape[0], dtype=np.float32)
 
     for i in range(norm_pts.shape[0]):
         s = np.random.uniform(0., 1., 2)
-        norm = norm_pts[i]
-        light = map_to_hemisphere(s, norm)
+        norm = np.array([0, 1, 0])
```
Comment (LapshinA) [inline utils/utils.py:287]: (0, 1, 0) - which axis do we use here? y-axis?

File: integrator.py (hunk 3 above)
Comment (givinar) [reply integrator.py:235]: In this case, yes, since the loss functions use the inverse determinant. This'll need to be fixed

File: integrator.py (hunk 4 above)
Comment (givinar) [reply integrator.py:225]: At the moment, it doesn’t matter. We'll optimize after obtaining the correct results.

File: utils/utils.py (hunk 5 above)
Comment (givinar) [reply utils/utils.py:287]: This should be clarified with Sergei.

File: integration_experiment.py (hunk 2 above)
Comment (givinar) [reply integration_experiment.py]: fixed
